docs: add `nuxtApp.runWithContext`

fimion opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)

Please carefully read the contribution docs before creating a pull request
 👉 https://nuxt.com/docs/community/contribution
-->

### 🔗 Linked issue

<!-- Please ensure there is an open issue and mention its number as #123 -->
#14723

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

@danielroe suggested [I take a crack at this](https://twitter.com/danielcroe/status/1703433066247565323) as there was not any public documentation about `callWithNuxt`. 

I've added a header stating that this method should be used sparingly, and that ultimately this is something that would ideally be solved at the framework level.

I've also included the [detailed explanation](https://github.com/nuxt/nuxt/issues/14269#issuecomment-1397352832) with edits that make more sense for documentation.

Let me know if there is any further details I should add, or if the format needs to change at all.

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

fimion committed:
docs: create call-with-nuxt

fimion committed:
docs: added details about callWithNuxt

danielroe reviewed (commented):
This is absolutely beautiful work.

The only thing I'd suggest is that we document `nuxtApp.runWithContext(() => fn())` instead, as it's a bit more intuitive how it works, and I think largely replaces `callWithNuxt`.

Very little would need to change for us to do this, just the function signature...

fimion committed:
docs: moved to use-nuxt-app

fimion commented:
Alright, I've moved it into the `useNuxtApp` docs and reworked the signature.

Are y'all using details/summary to collapse extra info? or would the detailed context documentation be better in a separate file under [Advanced](docs/3.api/4.advanced)?

fimion committed:
Merge branch 'main' into docs/call-with-nuxt

danielroe committed:
chore: small casing + spacing updates

danielroe committed:
docs: add docs for `asyncContext`

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

danielroe commented:
I think this is great, and as it's in the API section I think we can include all of this there. It's possible we could add something less/abbreviated in future in **Concepts**, maybe within https://nuxt.com/docs/guide/concepts/vuejs-development, or possibly a bit longer in a **Composables** section in  **Get Started**.
